src/Package/Impl/DataInspect/VariableGridHost.xaml.cs

```diff
@@ -54,20 +55,24 @@ public partial class VariableGridHost : UserControl {
         internal void SetEvaluation(VariableViewModel wrapper) {
             VsAppShell.Current.AssertIsOnMainThread();
 
+            // Is the variable gone?
             if (wrapper.TypeName == null) {
-                // the variable should have been removed
                 SetError(string.Format(CultureInfo.InvariantCulture, Package.Resources.VariableGrid_Missing, wrapper.Expression));
-            } else if (_evaluation == null
-                || (wrapper.Dimensions.Count == 2 && (wrapper.Dimensions[0] != _evaluation.Dimensions[0] || wrapper.Dimensions[1] != _evaluation.Dimensions[1]))) {
-                // matrix size changed. Reset the evaluation
-                ClearError();
-                VariableGrid.Initialize(new GridDataProvider(wrapper));
-                _evaluation = wrapper;
-            } else {
-                ClearError();
-                // size stays same. Refresh
+                _evaluation = null;
+                return;
+            }
+
+            ClearError();
+
+            // Does it have the same size and shape? If so, can update in-place (without losing scrolling etc).
+            if (_evaluation?.Dimensions.SequenceEqual(wrapper.Dimensions) == true) {
```

Comment:
Because of `?.`, we get a nullable bool here, so it has to be either ` == true` or `?? false` to get it to compile.